source/extensions/config_subscription/grpc/grpc_stream.h

```diff
@@ -259,6 +260,10 @@ class GrpcStream : public GrpcStreamInterface<RequestProto, ResponseProto>,
   const bool rate_limiting_enabled_;
   Event::TimerPtr drain_request_timer_;
 
+  // A stream value to be set in the control_plane.connected_state gauge once
+  // the gRPC-stream is establishing a connection or connected to the server.
+  uint32_t connected_state_val_;
```

Comment:
Given there are only a few of these, would using the "text readout" stat type be clearer rather than magic numbers?

Reply:
We can add a text-readout for the connected source (I'd say that it should be the gRPC-address or Envoy Cluster name), but I'm not sure it will add much more given that per-xDS-type has a [specific readout](https://github.com/envoyproxy/envoy/blob/69ee8cf7b5f5d6843d2f20c0149d3f2e55ea9e66/source/extensions/config_subscription/grpc/grpc_mux_impl.cc#L330).
IMHO it is easier to build alerts on top of integral values rather than text, but LMK if you think otherwise.

Reply:
I guess integers are fine as long as they aren't too magical. Please add documentation for this new stat and its values, as they are now a permanent-ish contract.

Reply:
Yeah, all the docs (including the stats docs) will be added in the next PR (after this and #35692 are merged).
The plan is to also flip the runtime flag to `true` after we run it internally for a few weeks and ensuring it all works well.

Reply:
FWIW, the docs describing the xDS-Failover feature are in #35746 (which is currently in draft mode until this is merged).

Reply:
I've changed the type from an unsigned int to an enum.
I believe the code is more understandable this way (but maybe less readable with the explicit template types). LMK if you think I should revert this.